### ISSUE TITLE
group.cpp:173: [realm-core-0.92.1] Assertion failed: top_size == 3 || top_size == 5 || top_size == 7

### ISSUE THREAD

**Issue by tspecht** [realm/realm-cocoa#2477]
Hi Guys!

I'm currently using the 0.95 release (Objective-C) and ran into the following issue today. After a restart the realm was empty but worked again .... Any ideas?

I was able to reproduce the issue by simply incrementing the version number on my realm configuration (e.g. when doing a migration). Here's the code that sets up the migration:

```
RLMRealmConfiguration* configuration = [RLMRealmConfiguration defaultConfiguration];
configuration.schemaVersion = 40;
configuration.migrationBlock = migrationBlock;
[RLMRealmConfiguration setDefaultConfiguration:configuration];
    
    // Open all the realms once to migrate them
   // Note: realmA-realmE are custom implemented methods which get the default configuration, set another path (without storing it as the default configuration again) and then return an instance created with RLMRealm realmWithConfiguration:
    [RLMRealm realmA];
    [RLMRealm realmB];
    [RLMRealm realmC];
    [RLMRealm realmD];
    [RLMRealm realmE];
```

And heres the exact crashlog:
```
group.cpp:173: [realm-core-0.92.1] Assertion failed: top_size == 3 || top_size == 5 || top_size == 7
0   MyApp                            0x0000000102502388 _ZN5realm4util18terminate_internalERNSt3__118basic_stringstreamIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 40
1   MyApp                            0x00000001024a185d _ZN5realm4util9terminateEPKcS2_l + 473
2   MyApp                            0x00000001025b5e24 _ZN5realm5Group6attachEm + 708
3   MyApp                            0x00000001025bf009 _ZN5realm11SharedGroup13do_begin_readENS0_9VersionIDE + 329
4   MyApp                            0x00000001025bddbe _ZN5realm11SharedGroup10begin_readENS0_9VersionIDE + 46
5   MyApp                            0x00000001024eebb4 _ZN5realm11SharedGroupC2ERNS_11ReplicationENS0_15DurabilityLevelEPKc + 1020
6   MyApp                            0x00000001024e949e -[RLMRealm initWithPath:key:readOnly:inMemory:dynamic:error:] + 1634
7   MyApp                            0x00000001024ea3c9 +[RLMRealm realmWithConfiguration:error:] + 721
8   MyApp                            0x00000001024e9c84 +[RLMRealm defaultRealm] + 71
9   MyApp                            0x00000001024b507a +[RLMObject allObjects] + 38
10  MyApp                            0x0000000101b95db5 __59-[DSAppDelegate application:didFinishLaunchingWithOptions:]_block_invoke_6 + 85
11  libdispatch.dylib                   0x0000000108e78fad _dispatch_call_block_and_release + 12
12  libdispatch.dylib                   0x0000000108e994bb _dispatch_client_callout + 8
13  libdispatch.dylib                   0x0000000108e81d3f _dispatch_root_queue_drain + 1829
14  libdispatch.dylib                   0x0000000108e81615 _dispatch_worker_thread3 + 111
15  libsystem_pthread.dylib             0x00000001091e1637 _pthread_wqthread + 729
16  libsystem_pthread.dylib             0x00000001091df40d start_wqthread + 13
IMPORTANT: if you see this error, please send this log to help@realm.io.
```

**Comment by jpsim** [comment]
Thanks for reporting this, @tspecht. We've gotten another report of this issue since our 0.95 release and are tracking this in our core repo issue tracker (https://github.com/realm/realm-core/issues/1092).

I don't think your repro case contains enough information for us to reproduce, but I'll give it a shot anyway and report back.

**Comment by jpsim** [comment]
@tspecht we believe this may have been fixed in 0.96.3. Can you try again?

**Comment by jpsim** [comment]
@tspecht have you seen this assertion failure lately?

**Comment by jpsim** [comment]
This was supposedly fixed in realm/realm-core#1892
